File: examples/lh-server/python/16-user-tasks/main.py
import asyncio
import sys

import littlehorse
from littlehorse.config import LHConfig
from littlehorse.model import PutUserTaskDefRequest, UserTaskField, VariableType
from littlehorse.worker import LHTaskWorker
from tasks import report_favorite_player
from workflow import get_workflow


def register(config: LHConfig) -> None:
    config.stub().PutUserTaskDef(
        PutUserTaskDefRequest(
            name="report-favorite-player",
            fields=[
                UserTaskField(
                    name="favorite-team",
                    display_name="Favorite Team",
                    type=VariableType.STR,
                    required=True,
                ),
                UserTaskField(
                    name="favorite-player-number",
                    display_name="Favorite Player's Number",
                    type=VariableType.INT,
                    required=True,
                ),
            ],
        )
    )
    littlehorse.create_task_def(
        report_favorite_player, "report-favorite-player", config
    )
    littlehorse.create_workflow_spec(get_workflow(), config)


async def workers(config: LHConfig) -> None:
    await littlehorse.start(
        LHTaskWorker(report_favorite_player, "report-favorite-player", config)
    )


if __name__ == "__main__":
    config = LHConfig()
    if len(sys.argv) != 2 or sys.argv[1] not in {"register", "workers"}:
        raise SystemExit("Usage: main.py [register|workers]")
    register(config) if sys.argv[1] == "register" else asyncio.run(workers(config))

File: examples/lh-server/go/16-user-tasks/main.go
package main

import (
	"context"
	"log"

	"github.com/littlehorse-enterprises/lh-developer-hub/examples/lh-server/go/internal/examplesupport"
	"github.com/littlehorse-enterprises/littlehorse/sdk-go/lhproto"
	"github.com/littlehorse-enterprises/littlehorse/sdk-go/littlehorse"
)

func main() {
	config := littlehorse.NewConfigFromEnv()
	client, err := config.GetGrpcClient()
	if err != nil {
		log.Fatal(err)
	}
	_, err = (*client).PutUserTaskDef(context.Background(), &lhproto.PutUserTaskDefRequest{
		Name: "report-favorite-player",
		Fields: []*lhproto.UserTaskField{
			{Name: "favoriteTeam", DisplayName: "Favorite Team", Type: lhproto.VariableType_STR, Required: true},
			{Name: "favoritePlayerNumber", DisplayName: "Favorite Player Number", Type: lhproto.VariableType_INT, Required: true},
		},
	})
	if err != nil {
		log.Fatal(err)
	}

	worker := examplesupport.NewWorker(config, reportFavoritePlayer, "report-favorite-player")
	workflow := littlehorse.NewWorkflow(wfLogic, "favorite-player-demo")
	examplesupport.Start(config, []*littlehorse.LHTaskWorker{worker}, workflow)
}
